gpt_researcher/agent.py

```
@@ -389,24 +436,81 @@ async def write_report(self, existing_headers: list = [], relevant_written_conte
+
+        Returns:
+            The generated introduction text.
+        """
         await self._log_event("research", step="writing_introduction")
         intro = await self.report_generator.write_introduction()
         await self._log_event("research", step="introduction_completed")
         return intro
 
-    async def quick_search(self, query: str, query_domains: list[str] = None) -> list[Any]:
-        return await get_search_results(query, self.retrievers[0], query_domains=query_domains)
+    async def quick_search(self, query: str, query_domains: list[str] = None, aggregated_summary: bool = False) -> list[Any] | str:
```

Ruff (0.14.14)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)